--- ./src/Imaging/DCRawFile.cs.old	2008-12-04 00:42:34.000000000 +0100
+++ ./src/Imaging/DCRawFile.cs	2008-12-04 00:42:44.000000000 +0100
@@ -90,7 +90,7 @@
 		internal static System.IO.Stream RawPixbufStream (Uri location)
 		{
 			string path = location.LocalPath;
-			string [] args = new string [] { dcraw_command, "-h", "-w", "-c", "-t", "0", path };
+			string [] args = new string [] { dcraw_command, "-h", "-w", "-c", "-t", "0", "-T", path };
 			
 			InternalProcess proc = new InternalProcess (System.IO.Path.GetDirectoryName (path), args);
 			proc.StandardInput.Close ();
