diff -ru gnome-do-plugins-0.8.0.old/configure.ac gnome-do-plugins-0.8.0/configure.ac
--- gnome-do-plugins-0.8.0.old/configure.ac	2009-01-30 03:53:45.000000000 +0100
+++ gnome-do-plugins-0.8.0/configure.ac	2009-01-30 04:21:14.000000000 +0100
@@ -31,7 +31,6 @@
 SHAMROCK_EXPAND_DATADIR
 
 PKG_CHECK_MODULES([BANSHEE_INDEXER], [banshee-1-collection-indexer >= 1.4.2])
-PKG_CHECK_MODULES([EVOLUTION_SHARP], [evolution-sharp])
 PKG_CHECK_MODULES([FLICKRNET], [flickrnet])
 PKG_CHECK_MODULES([GCONF_SHARP_20], [gconf-sharp-2.0])
 PKG_CHECK_MODULES([GLADE_SHARP_20], [glade-sharp-2.0])
@@ -56,7 +55,6 @@
 GDATA_SHARP_CONTACTS_LIBS="-r:$top_srcdir/BundledLibraries/Google.GData.Client.dll -r:$top_srcdir/BundledLibraries/Google.GData.Extensions.dll -r:$top_srcdir/BundledLibraries/Google.GData.Contacts.dll"
 
 AC_SUBST([BANSHEE_INDEXER_LIBS])
-AC_SUBST([EVOLUTION_SHARP_LIBS])
 AC_SUBST([FLICKRNET_LIBS])
 AC_SUBST([GCONF_SHARP_20_LIBS])
 AC_SUBST([GDATA_SHARP_ACL_LIBS])
@@ -121,7 +119,6 @@
 DiskMounter/Makefile
 EOG-Slideshow/Makefile
 Epiphany/Makefile
-Evolution/Makefile
 File/Makefile
 Firefox/Makefile
 Flickr/Makefile
Only in gnome-do-plugins-0.8.0: configure.ac.orig
diff -ru gnome-do-plugins-0.8.0.old/Makefile.am gnome-do-plugins-0.8.0/Makefile.am
--- gnome-do-plugins-0.8.0.old/Makefile.am	2009-01-30 03:53:45.000000000 +0100
+++ gnome-do-plugins-0.8.0/Makefile.am	2009-01-30 04:21:27.000000000 +0100
@@ -16,7 +16,6 @@
 	DiskMounter \
 	EOG-Slideshow \
 	Epiphany \
-	Evolution \
 	File \
 	Firefox \
 	Flickr \
