From 6fc9f57848f138db61b9e374ce4306305f556a77 Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Tue, 16 Jun 2009 15:48:16 +0000
Subject: Forgot one use in libgail

---
diff --git a/modules/other/gail/gailtreeview.c b/modules/other/gail/gailtreeview.c
index d615a45..3cdde65 100644
--- a/modules/other/gail/gailtreeview.c
+++ b/modules/other/gail/gailtreeview.c
@@ -867,7 +867,7 @@ gail_tree_view_ref_child (AtkObject *obj,
   gtk_tree_view_column_cell_set_cell_data (tv_col, tree_model, &iter, 
                                            is_expander, is_expanded);
 
-  renderer_list = gtk_tree_view_column_get_cell_renderers (tv_col);
+  renderer_list = gtk_cell_layout_get_cells (GTK_CELL_LAYOUT (tv_col));
 
   /* If there are more than one renderer in the list, make a container */
 
--
cgit v0.8.2

