--- work/gnome-desktop-2.29.92.old/configure.in	2010-03-21 16:10:17.000000000 +0000
+++ work/gnome-desktop-2.29.92/configure.in	2010-03-21 16:13:36.000000000 +0000
@@ -152,23 +152,47 @@
 
 dnl Checks for X's RANDR extension
 
-if $PKG_CONFIG --atleast-version $XRANDR_REQUIRED xrandr; then
-  echo "Building with the RANDR extension"
+AC_ARG_ENABLE(xrandr,
+  [AS_HELP_STRING([--enable-xrandr=(yes|no|auto)], [Enables or disables xrandr support (defaulr auto)])],
+  [ if test "x$enableval" = "xyes"; then
+      enable_xrandr=yes
+    elif test "x$enableval" = "xno"; then
+      enable_xrandr=no
+    elif test "x$enableval" = "xauto"; then
+      enable_xrandr=
+    else
+      echo "Unknown argument --enable-xrandr=$enableval"
+      exit 2
+    fi
+  ], [])
+
+if test x"$enable_xrandr" = x; then
+  PKG_CHECK_MODULES(XRANDR, [xrandr >= $XRANDR_REQUIRED], have_randr=yes, have_randr=no)
+elif test x"$enable_xrandr" = xyes; then
+  PKG_CHECK_MODULES(XRANDR, xrandr >= $XRANDR_REQUIRED)
+  have_randr=yes
+else
+  have_randr=no
+fi
+
+if test x$have_randr = xyes; then
   AC_DEFINE(HAVE_RANDR, 1,
-            [Define if the xrandr-1.2 library is present])
-  with_randr=yes
+            [Define if the xrandr-$XRANDR_REQUIRED library is present])
   RANDR_PACKAGE=xrandr
 else
-  echo "***** WARNING: Building without RANDR support"
-  with_randr=no
+  AC_MSG_RESULT(no)
   RANDR_PACKAGE=
+  XRANDR_CFLAGS=
+  XRANDR_LDFLAGS=
 fi
 
 AC_SUBST(RANDR_PACKAGE)
 
 dnl pkg-config dependency checks
 
-PKG_CHECK_MODULES(GNOME_DESKTOP, gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED gconf-2.0 >= $GCONF_REQUIRED $STARTUP_NOTIFICATION_PACKAGE $RANDR_PACKAGE)
+PKG_CHECK_MODULES(GNOME_DESKTOP, gdk-pixbuf-2.0 >= $GDK_PIXBUF_REQUIRED gtk+-2.0 >= $GTK_REQUIRED glib-2.0 >= $GLIB_REQUIRED gio-2.0 >= $GLIB_REQUIRED gconf-2.0 >= $GCONF_REQUIRED $STARTUP_NOTIFICATION_PACKAGE)
+GNOME_DESKTOP_CFLAGS += XRANDR_CFLAGS
+GNOME_DESKTOP_LDFLAGS += XRANDR_LDFLAGS
 
 dnl for gnome-about
 AM_PATH_PYTHON
